### PR TITLE
Notification menu option

### PR DESCRIPTION
**Pull request by bsonntag**
Implements the side menu's "Notifications" option. (Issue https://github.com/subvisual/council/issues/145)
For now this sends the user to a notifications page that shows new discussions and discussions with new comments.

### CODE REVIEW

**Comment by bsonntag** [comment]
@azevedo-252 Please review.

File: app/assets/javascripts/app/components/notification_card/notification_card.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  function NotificationCardCtrl($state, Notification) {
+    var ctrl = this;
+
+    ctrl.isNewDiscussion = isNewDiscussion;
+    ctrl.isNewComment = isNewComment;
+
+    function isNewDiscussion() {
+      return ctrl.notification.kind === 'new_discussion';
```
Comment (azevedo-252) [inline app/assets/javascripts/app/components/notification_card/notification_card.js]: Not sure this knowledge should be here?
[Tell Don't Ask](https://pragprog.com/articles/tell-dont-ask) ;)

File: app/assets/javascripts/app/notifications/notifications.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    .module('council.notifications')
+    .controller('NotificationsCtrl', NotificationsCtrl);
+
+  function NotificationsCtrl($state, Notification) {
+    var ctrl = this;
+
+    Notification.findAll()
+      .then(setNotifications);
+
+    function setNotifications(notifications) {
```
Comment (azevedo-252) [inline app/assets/javascripts/app/notifications/notifications.js]: Let's not use/maintain state in the controller. In this particular case you're just reassigning the resolved data intro a controller property. You could easily need to do some massaging with the resolved data, which would make you end up with unrelated logic inside you controller.
So as a rule of thumb, we don't maintain state in the controller, we actually resolve the data inside the service/factory itself, where it will manage it however it wants and then the controller simply requests the data when it's ready.
So, something like this:
```
Notification.findAll()
  .then(setNotifications);

function setNotifications() {
  ctrl.notifications = Notification.getAll();
}
```

File: app/assets/javascripts/app/notifications/notifications.module.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        template: '<div ui-view class="u-fullSize"></div>'
+      })
+      .state('notifications.index', {
+        url: '',
+        views: {
+          '': {
+            controller: 'NotificationsCtrl as ctrl',
+            templateUrl: 'notifications/index.html'
+          },
+          'sidebar@notifications.index': {
```
Comment (azevedo-252) [inline app/assets/javascripts/app/notifications/notifications.module.js]: why is this necessary?

File: app/assets/stylesheets/components/_icons.css.sass [hunk 4]
```diff
@@ -22,18 +22,12 @@
...
-    height: 20px
-    line-height: 20px
-    position: absolute
-    right: -35%
-    text-align: center
-    width: 20px
+.IconNavigation-count
+  @include delta
+  position: absolute
+  right: -35%
```
Comment (azevedo-252) [inline app/assets/stylesheets/components/_icons.css.sass]: What's -35%?

File: app/assets/stylesheets/components/_icons.css.sass [hunk 5]
```diff
@@ -22,18 +22,12 @@
...
+  bottom: 0
+  display: block
```
Comment (azevedo-252) [inline app/assets/stylesheets/components/_icons.css.sass]: Is this necessary?

File: app/assets/stylesheets/components/_notification_card.css.sass [hunk 6]
```diff
@@ -0,0 +1,23 @@
+.NotificationCard
+  position: relative
+
+  display: flex
+  flex-direction: column
+  min-height: 62px
```
Comment (azevedo-252) [inline app/assets/stylesheets/components/_notification_card.css.sass]: magic number

File: app/assets/javascripts/app/components/notification_card/notification_card.js (hunk 1 above)
Comment (bsonntag) [inline app/assets/javascripts/app/components/notification_card/notification_card.js]: Right. Moved to notification model.

File: app/assets/javascripts/app/notifications/notifications.js (hunk 2 above)
Comment (bsonntag) [inline app/assets/javascripts/app/notifications/notifications.js]: Ok. Done.

File: app/assets/javascripts/app/notifications/notifications.module.js (hunk 3 above)
Comment (bsonntag) [inline app/assets/javascripts/app/notifications/notifications.module.js]: It's not. It was from a previous attempt of trying a sidebar, and I forgot to remove this.
I removed it now.

**Comment by azevedo-252** [comment]
Don't forget the other comments I made and you didn't answer ;)

File: app/assets/stylesheets/components/_icons.css.sass (hunk 4 above)
Comment (bsonntag) [inline app/assets/stylesheets/components/_icons.css.sass]: I don't know. It was already there.

File: app/assets/stylesheets/components/_icons.css.sass (hunk 5 above)
Comment (bsonntag) [inline app/assets/stylesheets/components/_icons.css.sass]: No. Removed.

**Comment by azevedo-252** [comment]
:+1:
